subprojects/publish/src/main/java/org/gradle/api/publish/internal/GradleModuleMetadataWriter.java

```diff
@@ -259,15 +259,15 @@ private void writeVariants(PublicationInternal publication, SoftwareComponent co
         }
     }
 
-    private void writeCreator(JsonWriter jsonWriter) throws IOException {
+    private void writeCreator(PublicationInternal publication, JsonWriter jsonWriter) throws IOException {
         jsonWriter.name("createdBy");
         jsonWriter.beginObject();
         jsonWriter.name("gradle");
         jsonWriter.beginObject();
         jsonWriter.name("version");
         jsonWriter.value(GradleVersion.current().getVersion());
         jsonWriter.name("buildId");
-        jsonWriter.value(buildInvocationScopeId.getId().asString());
+        jsonWriter.value(publication.getBuildIdentifier().getOrElse(buildInvocationScopeId.getId().asString()));
```

Comment:
I thought we should set this as `convention()` where the property is created. But maybe we don't have the `buildInvocationScopeId` there... and it would also be the same code in two places. So this is probably fine.

Reply:
Yes, I thought about it as well, then decided against it because it was much more complicated in the end.